pkg/mpc-examples/test/01-WeightChanger.test.ts

```diff
@@ -0,0 +1,222 @@
+    });
+
+    it('Reweight duration set at 7 days', async () => {
+      assert.equal(await weightChanger.getReweightDuration(), time.WEEK);
+    });
+  });
+
+  // All weights are in terms of WETH/USDC
+  describe('Change Managed Pool weights', () => {
+    context('Change weights to 50/50', () => {
+      let desiredWeights: BigNumber[];
+      beforeEach('call make5050 function', async () => {
+        desiredWeights = toNormalizedWeights([fp(50), fp(50)]);
+        weightChanger = await deployController(deployer);
+        await weightChanger.make5050();
+      });
+
+      it('Failure after 4 days', async () => {
+        await fastForward(time.DAY * 4);
+        assert.isNotTrue(await checkTokenWeights(await weightChanger.getCurrentWeights(), desiredWeights));
+      });
+
+      it('Successful after 8 days', async () => {
+        await fastForward(time.DAY * 8);
+        assert.isTrue(await checkTokenWeights(await weightChanger.getCurrentWeights(), desiredWeights));
+      });
+    });
+
+    context('Change weights to 80/20', () => {
+      let desiredWeights: BigNumber[];
+      beforeEach('call make8020 function', async () => {
+        desiredWeights = toNormalizedWeights([fp(80), fp(20)]);
+        weightChanger = await deployController(deployer);
+        await weightChanger.make8020();
+      });
+
+      it('Failure after 4 days', async () => {
+        await fastForward(time.DAY * 4);
+        assert.isNotTrue(await checkTokenWeights(await weightChanger.getCurrentWeights(), desiredWeights));
+      });
+
+      it('Successful after 8 days', async () => {
+        await fastForward(time.DAY * 8);
+        assert.isTrue(await checkTokenWeights(await weightChanger.getCurrentWeights(), desiredWeights));
+      });
+    });
+
+    context('Change weights to 99/01', () => {
+      let desiredWeights: BigNumber[];
+      beforeEach('call make9901 function', async () => {
+        desiredWeights = toNormalizedWeights([fp(99), fp(1)]);
+        weightChanger = await deployController(deployer);
+        await weightChanger.make9901();
+      });
+
+      it('Failure after 4 days', async () => {
+        await fastForward(time.DAY * 4);
+        assert.isNotTrue(await checkTokenWeights(await weightChanger.getCurrentWeights(), desiredWeights));
+      });
+
+      it('Successful after 8 days', async () => {
+        await fastForward(time.DAY * 8);
+        assert.isTrue(await checkTokenWeights(await weightChanger.getCurrentWeights(), desiredWeights));
+      });
+    });
+  });
+
+  async function checkTokenWeights(_tokenWeights: BigNumber[], _desiredWeights: BigNumber[]): Promise<boolean> {
+    const tokenCount = (await weightChanger.getTokens()).length;
+    let correctWeights = 0;
+
+    for (let i = 0; i < tokenCount; i++) {
+      if (_desiredWeights[i].toString() === _tokenWeights[i].toString()) {
+        correctWeights += 1;
+      }
+    }
+    return correctWeights == tokenCount ? true : false;
+  }
```

Comment:
add the `Factory Access Control` tests that are in NullController tests: https://github.com/orbcollective/mpc-examples/blob/main/pkg/mpc-examples/test/00-NullController.test.ts#L151-L160